src/common/state/DBOptionsAttributes.C

```diff
@@ -1300,7 +1300,13 @@ DBOptionsAttributes::SetInt(const std::string &name, int defaultValue)
         optInts.push_back(defaultValue);
     }
     else
-        optInts[bIndex] = defaultValue;
+    {
+        OptionType ot = GetType(name);    
+        if (ot == Int)
+            optInts[bIndex] = defaultValue;
+        else if (ot == Enum)
+            optEnums[bIndex] = defaultValue;
```

Comment:
Maybe so. But not for this effort. Original code for setting enums doesn't do that error checking either.

Reply:
OMG...that comment you are responding to I wrote way back in May when you first created this PR. I know I didn't write it `1 hours ago` which is what GitHub's interface is showing.

I think what happend is that I *started* a review then and entered that comment. But, I never *finished* the review so that comment sat there until just today when I did finish it.